File: tests/fixtures/languages/go/animals.go
package animals

import (
	"fmt"
	"strings"
)

type Animal interface {
	Speak() string
}

type Dog struct {
	Name string
}

func (d *Dog) Speak() string {
	return fmt.Sprintf("%s says woof", d.Name)
}

func NewDog(name string) *Dog {
	d := &Dog{Name: strings.ToUpper(name)}
	return d
}

File: tests/fixtures/languages/rust/animals.rs
use std::fmt;
use std::collections::HashMap as Map;

pub trait Animal {
    fn speak(&self) -> String;
}

pub struct Dog {
    pub name: String,
}

impl Animal for Dog {
    fn speak(&self) -> String {
        self.bark()
    }
}

impl Dog {
    fn bark(&self) -> String {
        format!("{} says woof", self.name)
    }
}

fn make_dog(name: String) -> Dog {
    Dog { name }
}
